sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -1658,4 +1658,49 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
     val df = spark.createDataFrame(spark.sparkContext.makeRDD(rows), schema)
     assert(df.filter($"array1" === $"array2").count() == 1)
   }
+
+  test("SPARK-18207: Compute hash for wider table") {
+    import org.apache.spark.sql.types.{StructType, StringType}
```

Comment:
[This union](https://github.com/apache/spark/pull/15745/files/9346117b53fdefbcbc3391c8e05de066abb80811#diff-5d2ebf4e9ca5a990136b276859769289R1694) involves hashing. I confirmed that the original error occurred without this PR.